src/test/resources/features/Sessions.feature

```diff
@@ -0,0 +1,315 @@
+
+  @suite
+  Scenario Outline: Location based - User able to book a visit based on a location within an establishment where the Prisoner is based
+  # ensure a session restricted by location is only available to prisoners in that location
+
+    Given I navigate to VSIP service
+    And Im on "HMPPS Digital Services - Sign in" page
+    And I enter "<userName>"
+    And I enter the "<password>"
+    And click on submit button
+    Then Im on "Manage prison visits - Manage prison visits" page
+    And I select change establishment
+    And I change the establishmnet to Bristol
+    And click on continue button
+    And I click on Book a visit option
+    And Im on "Manage prison visits - Search for a prisoner" page
+    When I enter "<prisonNumber>" to search for a prison
+    And click on search button
+    Then I choose prison from search results
+    Then Im on "Manage prison visits - Vsip_prisoner12, Do Not Use" page
+    #Location of a prisoner is being checked in this step if they can't carry on means Location is changed.
+    And I check Prisoner location "A-2-007, Bristol (HMP)"
+    And I click on Book a visit button
+    Then Im on "Manage prison visits - Select visitors from the prisoner’s approved visitor list" page
+    And I select a visitr form the list from Bristol establishment
+    And click on continue button
+    Then Im on "Manage prison visits - Select date and time of visit" page
+    And I select time slot
+    And click on continue button
+    Then Im on "Manage prison visits - Is additional support needed for any of the visitors?" page
+    And I select No for additional support needed
+    And click on continue button
+    Then Im on "Manage prison visits - Who is the main contact for this booking?" page
+    And I select the main contact option
+    And I enter "<phoneNumber>" to get text message
+    And click on continue button
+    Then Im on "Manage prison visits - How was this booking requested?" page
+    And I select a phone call option on method used to make the reqest
+    And click on continue button
+    Then Im on "Manage prison visits - Check the visit details before booking" page
+    And click on continue button
+    #And click on submit booking button
+    Then Im on "Manage prison visits - Booking confirmed" page
+    And I see "Booking confirmed" message displayed
+    And I see a booking reference
+    And I sign out of the service
+    #Reverting the Booking for consistent Test execution
+    And I want to clean up after the above test
+
+    Examples:
+      | userName  | password          | prisonNumber | phoneNumber     |
+      | VSIP4_TST | DigitalServices10 | A8897DZ      | 07806156987     |
+
+  @suite
+  Scenario Outline: Location not based - User not able to book a visit based on a location within an establishment where the Prisoner is based
+  # ensure a session restricted by location is not available to a prisoner not in that location
```

Comment:
this is a handy test but as discussed this morning it cannot be used for location based tests since no slots will be available to the prisoner as they are in a temporary location. A prisoner needs to be chosen that is in a different location than the previous test and make sure that each others slots are not available to one another.